llvm/cmake/modules/AddOCaml.cmake

```diff
@@ -38,12 +38,20 @@ function(add_ocaml_library name)
   set(ocaml_inputs)
 
   set(ocaml_outputs "${bin}/${name}.cma")
+  # Always use -custom when building in-tree
```

Comment:
Why do we always use -custom for in-tree builds? Based on https://ocaml.org/manual/4.10/intfc.html#ss:staticlink-c-code, do I correctly understand that you need -custom if you're linking statically? Wasn't the previous BUILD_SHARED_LIBS based condition correct for that?